Automattic/wp-calypso
# Block Editor: Update the Convert to Blocks dialog

#### Changes proposed in this Pull Request

* Update the copy of the Convert to Blocks dialog.
* Update the modal to use the Core component.
* Add a new persistent `ui.convertToBlocksDialog` state that keeps track of dismissed Convert to Blocks dialogs, in order to only show it once per post.

| Before | After |
| - | - |
| <img width="617" alt="Image 2020-05-27 at 9 29 53 AM" src="https://user-images.githubusercontent.com/5634774/83024613-a52d3600-9ffc-11ea-881d-c2d182a5f4ff.png"> | <img width="735" alt="Screenshot 2020-06-02 at 11 25 02" src="https://user-images.githubusercontent.com/2070010/83510099-597b0080-a4c4-11ea-972f-97bfa852e442.png"> |

#### Testing instructions

* On a site using the Classic Editor, create a new post and write some content.
* Switch over to Gutenberg (Gutenframe).
* Make sure the "Convert to Blocks" notice shows up.
* Dismiss it (don't convert to blocks!) and reload the editor.
* Make sure the notice doesn't show up anymore.
* Switch back to Classic editor, and create another post with some content.
* Switch to Gutenberg, and make sure the notice shows up again.

Note: after switching to Gutenberg, adding **any** block content (including an empty Paragraph block) will automagically disable the notice as well.

Fixes https://github.com/Automattic/dotcom-manage/issues/87
